docs/madr/decisions/005-policy-matching.md

```diff
@@ -0,0 +1,1052 @@
+* MeshGatewayRoute – policy affects only a specified MeshGatewayRoute (select route using "name")
+* MeshHTTPRoute – policy affects only a specified MeshHTTPRoute (select resource using "name"), MeshHTTPRoute is not implemented at that moment
+
+The top-level "TargetRef" identifies a set of DPPs that will be affected by the policy.
+
+Traffic direction is expressed using "to" and "from" arrays. 
+
+If policy configures outbounds, then configuration is specified in "to" array: 
+```yaml
+type: UpstreamTimeout
+mesh: mesh-1
+name: timeouts
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: payments
+      conf:
+        connectTimeout: 10s
+    - targetRef:
+        kind: MeshService
+        name: web-api
+      conf:
+        connectTimeout: 20s
+```
+
+If policy configures inbounds, then configuration is specified in "from" array: 
+```yaml
+type: MeshTrafficPermission
+mesh: mesh-1
+name: permissions
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  from:
+    - targetRef:
+        kind: Mesh
+        name: mesh-1
+      conf:
+        action: DENY
+    - targetRef:
+        kind: MeshServiceSubset
+        name: backend
+        tags:
+          version: v2
+      conf:
+        action: ALLOW
+    - targetRef:
+        kind: MeshSubset
+        name: mesh-1
+        tags:
+          kuma.io/zone: us-east
+      conf:
+        action: ALLOW
+```
+
+Also, there are policies that configure both inbounds and outbounds (like TrafficLog). 
+And there are policies that configure none of this (like ProxyTemplate).
+
+### Applying
+
+#### Outbound policy
+
+All outbound policies contains a "to" array.
+Each element of the array has a "targetRef" that selects outbound
+and has an outbound configuration:
+
+```yaml
+to:
+  - targetRef: Mesh | MeshService | MeshHTTPRoute
+    conf: ...
+```
+
+If a data plane proxy has 3 outbounds:
+```yaml
+outbound:
+  - port: 8081
+    tags:
+      kuma.io/service: backend
+  - port: 8082
+    tags:
+      kuma.io/service: web-api
+  - port: 8083
+    tags:
+      kuma.io/service: payments
+```
+
+then policy can select a single "web-api" outbound:
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshService
+      name: web-api
+    conf: ... # conf for "web-api" outbound
+```
+
+policy can select all outbounds ("backend", "web-api" and "payments"):
+
+```yaml
+to:
+  - targetRef:
+      kind: Mesh
+      name: mesh-1
+    conf: ... # conf for all outbounds
+```
+
+policy can select an MeshHTTPRoute:
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshHTTPRoute
+      name: payments-get-info-v1-httproute
+    conf: ... # conf for "/v1/getinfo/" route of "payments" outbound 
+```
+
+When several targetRefs select the same outbound,
+corresponding confs are merged in the order they're presented in the "to" array 
+(bottom ones have more priority over the top ones):
```

Comment:
I'm not a native speaker but things can **have** "higher priority" and someone / something can **give** "more priority" to things. Maybe native speakers can chip in - @michaelbeaumont @parkanzky 

```suggestion
(bottom ones have higher priority over the top ones):
```